.travis.yml

```diff
@@ -1,15 +1,21 @@
 language: julia
 julia:
   - 0.6
+  - 0.7
+  - 1.0
   - nightly
+
+codecov: true
+coveralls: true
+
 notifications:
     email: false
 sudo: false
-script:
-  - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-  - julia --check-bounds=yes -e 'Pkg.clone(pwd()); Pkg.test("Coverage"; coverage=true)'
-  - julia --code-coverage=user -e 'cd(Pkg.dir("Coverage")); using Coverage; cov_res = process_folder(); Coveralls.submit(cov_res); Codecov.submit(cov_res)';
-after_success:
-  - if [ "$TRAVIS_JULIA_VERSION" != "nightly" ]; then
-      julia -e 'cd(Pkg.dir("Coverage")); using Coverage; cov_res = process_folder(); Coveralls.submit(cov_res); Codecov.submit(cov_res)';
-    fi
```

Comment:
Travis' `after_success` block runs after the `script` section has passed successfully, i.e. with an exit code (in the shell sense) of 0. Most packages submit coverage to Codecov and/or Coveralls in the `after_success` blocks. We're doing the same here, since we don't actually _submit_ coverage results during the `script`. Does that sort of make sense?

If you're asking about the particular code, i.e.
```julia
using Coverage
cov_res = process_folder()
Coveralls.submit(cov_res)
Codecov.submit(cov_res)
```
that's processing the package's folder once, then submitting to Coveralls, then submitting to Codecov.

Reply:
What do you man, we "don't actually submit coverage results during the `script`" ? My point here is that we *do*: the command
```
julia -e 'cd(Pkg.dir("Coverage")); using Coverage; cov_res = process_folder(); Coveralls.submit(cov_res); Codecov.submit(cov_res)';
```
is run twice (at least in the old `.travis.yml`): once in `script:` and once in `after_success:`. I see nothing in the former that would turn off actual submission.
The only difference is in that in `script:`, the option `--code-coverage=user` is added, which is why I think it really actually is there to collect coverage data on the coveralls/codecov specific cod.

Reply:
Oh wow, I'm blind. I totally missed that in the old `script` section. :man_facepalming: Yeah, don't worry about that, it doesn't have to be run twice; just once in `after_script` is fine. The default script runs Julia with the appropriate arguments to produce coverage information for the package being tested.

Reply:
I did the test. Dropping the custom script causes a major drop in coverage, from ~73% to ~54%. So I am leaving this in for now.